File: HostApps/ios/ReactNative/TestNfcSwiftNew/ios/NfcCardModule.m
//
//  NfcCardModule.m
//  TestNfcSwiftNew
//
//  Created by Alina Alinovna on 14.06.2020.
//

#import <Foundation/Foundation.h>

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(NfcCardModule, NSObject)



/* Ton wallet applet stuff*/

RCT_EXTERN_METHOD(
  getSault: (RCTPromiseResolveBlock)resolve
  rejecter: (RCTPromiseRejectBlock)reject
)
RCT_EXTERN_METHOD(
  getTonAppletState: (RCTPromiseResolveBlock)resolve
  rejecter: (RCTPromiseRejectBlock)reject
)


@end

File: HostApps/ios/ReactNative/TestNfcSwiftNew/ios/NfcCardModule.swift
import Foundation
import CoreNFC
import PromiseKit

@available(iOS 13.0, *)
@objc(NfcCardModule)
class NfcCardModule: NSObject {
  var nfcApi: TonNfcApi = TonNfcApi()
  
  
  @objc
  func getSault(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    nfcApi.setPromiseStuff(resolveBlock: resolve, rejectBlock: reject)
    nfcApi.getSault()
  }
  
  
  @objc
  func getTonAppletState(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    nfcApi.setPromiseStuff(resolveBlock: resolve, rejectBlock: reject)
    nfcApi.getTonAppletState()
  }
  
  

}

